compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -2800,7 +2800,7 @@ private VariableDefinitionNode createBLangVarDef(Location location,
                                                 Optional<io.ballerina.compiler.syntax.tree.ExpressionNode> initializer,
                                                 Optional<Token> finalKeyword) {
         BindingPatternNode bindingPattern = typedBindingPattern.bindingPattern();
-        BLangVariable variable = getBLangVariableNode(bindingPattern);
+        BLangVariable variable = getBLangVariableNode(bindingPattern, location);
```

Comment:
Related to the above comment. Looks like in the AST we either have to set typedBindingPattern pos or bindingPattern pos depending on the context. Not both positions. Which does not allow us to pick the preferred pos in later stages.

Reply:
I think what we are doing in the nodebuilder is correct, since for the variable we need to set the full length pos. When we are logging the error I think we need to calculate the exact pos with the help of typedesc pos because we cannot hold binding pattern pos in blang node.

Reply:
Yes, previously there were instances where incorrect positions were set to the whole var. This lead to issues in the Semantic API. Let me check and see whether these would be a simple fix. Can see some places where the var node pos is taken instead of the type node's pos.

Reply:
Shall we look into this separately? Just simply changing it to get the type node's position caused negative tests to fail in a different manner

Reply:
+1

Reply:
+1

Reply:
Created #34642 to track this.